lib/base/atomic.hpp

```diff
@@ -35,7 +35,7 @@ class Atomic : public std::atomic<T> {
 	 * @param desired Initial value
 	 * @param order Initial store operation's memory order
 	 */
-	inline Atomic(T desired, std::memory_order order)
+	inline Atomic(T desired, std::memory_order order) : std::atomic<T>(desired)
 	{
 		this->store(desired, order);
 	}
```

Comment:
What is this even supposed to achieve?

Like I can't imagine a scenario where this would make sense. Every other operation on that variable should be synchronized after its construction anyway. If it wasn't, that would mean there is another thread calling something like `load()` while this constructor hasn't finished and I don't see how this constructor would even improve anything here? I mean that `load()` could still happen before or while the `std::atomic` constructor is doing its work.

The only potentially valid reason I can currently imagine for `icinga::Atomic` would be to hide the strange pre-C++20 default constructor of `std::atomic` that would technically[^1] require `std::atomic_init`.

[^1]: At least practically, on my machine `atomic_init()` just seems to call `store()` itself, so I don't expect any observable difference.

Reply:
> Every other operation on that variable should be synchronized after its construction anyway.

OK, I believe you.